# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../ratpoison/ratpoison-1.4.5-staticgravity.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

http://minimyth.googlecode.com/svn/trunk/gar-minimyth/script/X11/ratpoison/files/ratpoison-1.4.5-staticgravity.patch
diff -Naur ratpoison-1.4.5-old/src/actions.c ratpoison-1.4.5-new/src/actions.c
--- ratpoison-1.4.5-old/src/actions.c	2009-07-14 19:56:16.000000000 -0700
+++ ratpoison-1.4.5-new/src/actions.c	2009-07-14 19:56:56.000000000 -0700
@@ -1882,6 +1882,8 @@
 {
   int ret = -1;
 
+  if (!strcasecmp (data, "static") || !strcasecmp (data, "static") || !strcmp (data, "10"))
+    ret = StaticGravity;
   if (!strcasecmp (data, "northwest") || !strcasecmp (data, "nw") || !strcmp (data, "7"))
     ret = NorthWestGravity;
   if (!strcasecmp (data, "north") || !strcasecmp (data, "n") || !strcmp (data, "8"))
diff -Naur ratpoison-1.4.5-old/src/manage.c ratpoison-1.4.5-new/src/manage.c
--- ratpoison-1.4.5-old/src/manage.c	2009-07-14 19:56:16.000000000 -0700
+++ ratpoison-1.4.5-new/src/manage.c	2009-07-14 19:56:56.000000000 -0700
@@ -541,6 +541,9 @@
     case SouthEastGravity:
       win->x = frame->x + frame->width - win->width - win->border;
       break;
+    case StaticGravity:
+      win->x = frame->x + win->x;
+      break;
     }
 
   /* Y coord. */
@@ -561,6 +564,9 @@
     case SouthWestGravity:
       win->y = frame->y + frame->height - win->height - win->border;
       break;
+    case StaticGravity:
+      win->y = frame->y + win->y;
+      break;
     }
 }
 
